bcda/auth/backend.go

```diff
@@ -261,22 +226,48 @@ func getPublicKey() *rsa.PublicKey {
 	return rsaPub
 }
 
-func getJWTClaims(backend *JWTAuthenticationBackend, tokenString string) jwt.MapClaims {
+func (backend *JWTAuthenticationBackend) GetJWTClaims(tokenString string) jwt.MapClaims {
+	token, err := backend.GetJWTToken(tokenString)
+
+	if err != nil {
+		panic(err)
+	}
+
+	if !token.Valid {
+		return nil
+	}
+
+	return token.Claims.(jwt.MapClaims)
+}
+
+func (backend *JWTAuthenticationBackend) GetJWTToken(tokenString string) (*jwt.Token, error) {
+
 	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
 		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
 			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
 		}
 
 		return backend.PublicKey, nil
 	})
+	return token, err
+}
 
+// Save a token to the DB for a user
+func (backend *JWTAuthenticationBackend) CreateToken(user bcdagorm.User) (bcdagorm.Token, error) {
+	db := database.GetGORMDbConnection()
+	tokenString, err := backend.GenerateTokenString(user.UUID.String(), user.Aco.UUID.String())
 	if err != nil {
 		panic(err)
 	}
-
-	if !token.Valid {
-		return nil
+	// Get the claims of the token to find the token ID that was created
+	claims := backend.GetJWTClaims(tokenString)
+	token := bcdagorm.Token{
+		UUID:   uuid.Parse(claims["id"].(string)),
+		UserID: user.UUID,
+		Value:  tokenString,
```

Comment:
One last request for me: please make sure we're hashing token values anywhere we are inserting/updating them in the database. Check out: https://github.com/CMSgov/bcda-app/blob/master/bcda/auth/backend.go#L135

Reply:
Resolved in latest commit.  token.Value is now hashed on save IF it is a valid JWT token.  If it isn't then I assume it is already hashed and don't want to double hash it.